vividus-plugin-kafka/src/test/java/org/vividus/bdd/steps/kafka/KafkaStepsIntegrationTests.java

```diff
@@ -87,14 +93,18 @@ void beforeEach()
 
         when(propertyParser.getPropertyValuesByPrefix("kafka.producer.")).thenReturn(producerConfigs);
         when(propertyParser.getPropertyValuesByPrefix("kafka.consumer.")).thenReturn(consumerConfigs);
-        kafkaSteps = new KafkaSteps(propertyParser, new SimpleTestContext(), bddVariableContext, softAssert);
+        kafkaSteps = new KafkaSteps(propertyParser, testContext, bddVariableContext, softAssert);
     }
 
     @Test
     void shouldProduceToAndConsumerFromKafka() throws InterruptedException, ExecutionException, TimeoutException
     {
         kafkaSteps.startKafkaListener(Set.of(TOPIC));
 
+        kafkaSteps.startKafkaListener(Set.of(TOPIC));
+
+        verify(testContext).remove(GenericMessageListenerContainer.class);
```

Comment:
it's better to have a separate unit test (not integration one) to cover the use case